view/src/client.rs

```diff
@@ -146,32 +145,18 @@ pub trait ViewClient {
     /// Queries for all known assets.
     fn assets(&mut self) -> Pin<Box<dyn Future<Output = Result<asset::Cache>> + Send + 'static>>;
 
-    /// Queries for transaction hashes in a range of block heights
-    fn transaction_hashes(
-        &mut self,
-        start_height: Option<u64>,
-        end_height: Option<u64>,
-    ) -> Pin<Box<dyn Future<Output = Result<Vec<(u64, Vec<u8>)>>> + Send + 'static>>;
-
-    /// Queries for a transaction hashes by its transaction hash
-
-    fn transaction_by_hash(
-        &mut self,
-        tx_hash: tendermint::hash::Hash,
-    ) -> Pin<Box<dyn Future<Output = Result<Option<Transaction>>> + Send + 'static>>;
-
     /// Generates a full perspective for a selected transaction using a full viewing key
-    fn transaction_perspective(
+    fn transaction_info_by_hash(
         &mut self,
-        tx_hash: tendermint::hash::Hash,
-    ) -> Pin<Box<dyn Future<Output = Result<TransactionPerspective>> + Send + 'static>>;
+        id: penumbra_transaction::Id,
+    ) -> Pin<Box<dyn Future<Output = Result<TransactionInfo>> + Send + 'static>>;
 
     /// Queries for transactions in a range of block heights
-    fn transactions(
+    fn transaction_info(
         &mut self,
         start_height: Option<u64>,
         end_height: Option<u64>,
-    ) -> Pin<Box<dyn Future<Output = Result<Vec<(u64, Transaction)>>> + Send + 'static>>;
+    ) -> Pin<Box<dyn Future<Output = Result<Vec<TransactionInfo>>> + Send + 'static>>;
```

Comment:
This could be nicer if it took a https://doc.rust-lang.org/std/ops/struct.Range.html

Reply:
never mind, it should really be a https://doc.rust-lang.org/std/ops/trait.RangeBounds.html but we can't use that because it would make the trait not be object safe